test/models/metrics/algorithms_test.rb

```diff
@@ -0,0 +1,147 @@
+
+  test 'isbns counts are included in monthly aggregation' do
+    aggregate = Metrics::Algorithms.new.generate(DateTime.now)
+    assert aggregate.isbn == 1
+  end
+
+  test 'pmids counts are included in monthly aggregation' do
+    aggregate = Metrics::Algorithms.new.generate(DateTime.now)
+    assert aggregate.pmid == 1
+  end
+
+  test 'unmatched counts are included are included in monthly aggregation' do
+    aggregate = Metrics::Algorithms.new.generate(DateTime.now)
+    assert aggregate.unmatched == 2
+  end
+
+  test 'creating lots of searchevents leads to correct data for monthly' do
+    # drop all searchevents to make math easier and minimize fragility over time as more fixtures are created
+    SearchEvent.delete_all
+
+    doi_expected_count = rand(1...100)
+    doi_expected_count.times do
+      SearchEvent.create(term: terms(:doi), source: 'test')
+    end
+
+    issn_expected_count = rand(1...100)
+    issn_expected_count.times do
+      SearchEvent.create(term: terms(:issn_1075_8623), source: 'test')
+    end
+
+    isbn_expected_count = rand(1...100)
+    isbn_expected_count.times do
+      SearchEvent.create(term: terms(:isbn_9781319145446), source: 'test')
+    end
+
+    pmid_expected_count = rand(1...100)
+    pmid_expected_count.times do
+      SearchEvent.create(term: terms(:pmid_38908367), source: 'test')
+    end
+
+    unmatched_expected_count = rand(1...100)
+    unmatched_expected_count.times do
+      SearchEvent.create(term: terms(:hi), source: 'test')
+    end
+
+    aggregate = Metrics::Algorithms.new.generate(DateTime.now)
+
+    assert doi_expected_count == aggregate.doi
+    assert issn_expected_count == aggregate.issn
+    assert isbn_expected_count == aggregate.isbn
+    assert pmid_expected_count == aggregate.pmid
+    assert unmatched_expected_count == aggregate.unmatched
+  end
+
+  # Total
+  test 'dois counts are included in total aggregation' do
+    aggregate = Metrics::Algorithms.new.generate
+    assert aggregate.doi == 1
+  end
+
+  test 'issns counts are included in total aggregation' do
+    aggregate = Metrics::Algorithms.new.generate
+    assert aggregate.issn == 1
+  end
+
+  test 'isbns counts are included in total aggregation' do
+    aggregate = Metrics::Algorithms.new.generate
+    assert aggregate.isbn == 1
+  end
+
+  test 'pmids counts are included in total aggregation' do
+    aggregate = Metrics::Algorithms.new.generate
+    assert aggregate.pmid == 2
+  end
+
+  test 'unmatched counts are included are included in total aggregation' do
+    aggregate = Metrics::Algorithms.new.generate
+    assert aggregate.unmatched == 2
+  end
+
+  test 'creating lots of searchevents leads to correct data for total' do
+    # drop all searchevents to make math easier and minimize fragility over time as more fixtures are created
+    SearchEvent.delete_all
+
+    doi_expected_count = rand(1...100)
+    doi_expected_count.times do
+      SearchEvent.create(term: terms(:doi), source: 'test')
+    end
```

Comment:
I'm not sure where to make this comment, but it's been gnawing at me and this was one of the places where it jumps into my brain.

I don't think we need to change anything - but I am noting that we've chosen to weight these totals by search frequency, rather than weighting every term in the collection equally. I'm not sure whether that decision is the right one - it seems fine to start with, and having reviewed the ADR for this I don't think we specifically discussed this question.

I'm commenting here mostly to get this thought out of my head, and see what either of you think about this. I'm fine with merging as is now and returning to this question down the road if needed.

Reply:
I think the behavior here is correct, assuming the question is "how often does tacos understand a given search event?"

If the question was instead, "what percentage of search terms does tacos understand?" then I could see flipping it. However, as TACOS is designed to work at the searchevent level I believe this is the correct question and correct metric. I'm not opposed to _also_ adding a variant metric to track Terms instead of Events.

Please let me know if I didn't understand your question.